src/proof/lrat/lrat_proof.cpp

```diff
@@ -0,0 +1,199 @@
+/*********************                                                        */
+/*! \file lrat_proof.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Alex Ozdemir
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief DRAT Proof Format
+ **
+ ** Defines deserialization for DRAT proofs.
+ **/
+
+#include "proof/lrat/lrat_proof.h"
+
+#include "base/output.h"
+#include "base/cvc4_assert.h"
+
+#include <algorithm>
+#include <cstdlib>
+#include <iostream>
+#include <fstream>
+#include <unordered_map>
+
+namespace CVC4 {
+namespace proof {
+namespace lrat {
+
+using prop::SatClause;
+using prop::SatLiteral;
+using prop::SatVariable;
+
+LRATProof::LRATProof(
+    const std::unordered_map<ClauseId, prop::SatClause*>& usedClauses,
+    const std::vector<ClauseId>& clauseOrder,
+    const std::string& dratBinary)
+{
+  std::ostringstream cmd;
+  char formulaFilename[] = "/tmp/cvc4-dimacs-XXXXXX";
+  char dratFilename[] = "/tmp/cvc4-drat-XXXXXX";
+  char lratFilename[] = "/tmp/cvc4-lrat-XXXXXX";
+  int r;
+  r = mkstemp(formulaFilename);
+  Assert(r > 0);
+  r = mkstemp(dratFilename);
+  Assert(r > 0);
+  r = mkstemp(lratFilename);
+  Assert(r > 0);
+  std::ofstream formStream(formulaFilename);
```

Comment:
I'd  like to avoid the temporary files, but  I'm not sure how to.

In particular, we're going to link against drat-trim, so even its interface supports receiving one of its two in puts on stdin, its stdin is our stdin, so I'm not sure if we can use that part of the interface.

I could spend some time figuring out memfd, but I suspect that would be hard to do cross-platform.

Reply:
In the meantime, I'll  close the fd produced by `mkstemp`.

Reply:
I didn't realize that we were going to link against drat trim. I was somehow assuming that we would execute it as a separate process. I think that the current solution is ok for now and we can try to make it file-free later